stringdtype/stringdtype/src/dtype.c

```diff
@@ -179,13 +179,32 @@ nonzero(void *data, void *NPY_UNUSED(arr))
 // Implementation of PyArray_CompareFunc.
 // Compares unicode strings by their code points.
 int
-compare_strings(char **a, char **b, PyArrayObject *NPY_UNUSED(arr))
+compare(void *a, void *b, void *NPY_UNUSED(arr))
 {
     ss *ss_a = (ss *)a;
     ss *ss_b = (ss *)b;
     return strcmp(ss_a->buf, ss_b->buf);
 }
 
+// PyArray_ArgmaxFunc
+// The max element is the one with the highest unicode code point.
+int
+argmax(void *data, npy_intp n, npy_intp *max_ind, void *arr)
+{
+    // Get current index and length to avoid indexing outside of the array
+    PyArrayObject *array = (PyArrayObject *)arr;
+    ss *arr_data = (ss *)PyArray_DATA(array);
+    int start_index = (arr_data - (ss *)data) / PyArray_ITEMSIZE(array);
+
+    *max_ind = start_index;
+    for (int i = start_index; i < n; i++) {
```

Comment:
The versions of these functions in numpy do:

```
*max_ind = 0
for (int i = 1; i < n; i++)
```

Why is this any different?

Reply:
https://github.com/numpy/numpy/blob/ea0b1708b27e683201285da53c6afda24bc8ba74/numpy/core/src/multiarray/arraytypes.c.src#L3572-L3600